components/script/dom/webglrenderingcontext.rs

```diff
@@ -502,6 +502,12 @@ impl WebGLRenderingContext {
             return;
         }
 
+        // Dirtying the canvas is unnecessary if we're actively displaying immersive
+        // XR content right now.
+        if self.global().as_window().in_immersive_xr_session() {
+            return;
```

Comment:
I'm a bit worried that when we come to put DOMToTexture nodes into XR sessions, we'll forget that we did this.